### ISSUE TITLE
Nightly test grpo-qwen3.5-35ba3b-2n8g-megatron-ep16 failing after the vllm bump PR (#2384)

### ISSUE THREAD

**Issue by kajalj22** [NVIDIA-NeMo/RL#2619]
**Describe the bug**

Nightly test grpo-qwen3.5-35ba3b-2n8g-megatron-ep16 failing after the vllm bump PR (https://github.com/NVIDIA-NeMo/RL/pull/2384) with a CUDA OOM 

**Steps/Code to reproduce bug**

Test failing off main branch

**Additional context**

The last successful run was on May 22nd, no test recorded before that.

**Comment by zpqiu** [comment]
One observation from the failing run: after vLLM enters sleep mode, the residual GPU memory reported by vLLM appears to grow across steps.

For example, in one run we saw:

- After initial sleep: `Sleep mode freed 43.82 GiB, 4.11 GiB memory is still in use`
- After Step 1: `11.57 GiB memory is still in use`
- After Step 2: around `25.0 GiB memory is still in use`
- Around Step 3: it stayed around `23-25 GiB memory is still in use`

The eventual OOM happened later in the Megatron training/logprob path, but the available memory was already reduced by the residual vLLM memory after generation/sleep.

For now, the only workaround we found is to move the MCore/Megatron path from TP=1 to TP=2, with `expert_tensor_parallel_size=1` and `sequence_parallel=True`. With that configuration, the local nightly
test passed.

**Comment by sharonyu-115** [comment]
Conducted the following comparison runs:

- 501cd12ce (immediately before #2384) - vLLM 0.17.1 / torch 2.10: Steps 1–3 clean, no OOM
- 241ece552 (#2384) - vLLM 0.20.0 / torch 2.11： OOM at step 2 

So it is exactly #2384 triggering the OOM to occur. 

Then did the comparison run between vLLM 0.17 and vLLM 0.20, to isolate it is not vLLM 0.20 keeps more memory resident after sleep, which would eat the headroom and cause the step-2 OOM. 

Single-shot  — resident still in use after sleep(level=1)
<img width="565" height="275" alt="Image" src="https://github.com/user-attachments/assets/15d2b3e5-86f7-4b0f-b6d1-90f8a1ce258f" />

 4-cycle probe — does sleep/wake leak across iterations?

<img width="412" height="129" alt="Image" src="https://github.com/user-attachments/assets/10cad1ff-9565-4fe6-9d49-41452c0a2d20" />

The tests concluded vLLM 0.20 is not the root cause. But since vLLM 0.20 pins to torch 2.11, this run cannot fully separate vLLM version from torch version. 

The OOM is at the step-2 logprob exp() (a separate fp32-logits transient). The dump shows the allocator is holding 5.90 GiB it isn't using: 

<img width="428" height="267" alt="Image" src="https://github.com/user-attachments/assets/15eb1cf3-5e77-48c5-b912-177e990c432b" />

Could be fragmentation related (torch 2.10→2.11 allocator behavior change?).  Tried and confirmed that setting `expandable_segments:True` can avoid the OOM error. 

Though root cause not yet identified, apply the configuration changes suggested to address the OOM issue for now. 

@zpqiu  @yuki-97  could you help review the changes? Thanks!

**Comment by yuki-97** [comment]
closed through https://github.com/NVIDIA-NeMo/RL/pull/2312
